builtin/providers/aws/resource_aws_instance.go

```diff
@@ -760,6 +760,10 @@ func fetchRootDeviceName(ami string, conn *ec2.EC2) (*string, error) {
 		rootDeviceName = image.BlockDeviceMappings[0].DeviceName
 	}
 
+	if rootDeviceName == nil {
+		return nil, fmt.Errorf("[WARN] Error finding Root Device Name for AMI (%s)", ami)
```

Comment:
@catsby ah okay. so if we can verify that these steps still work:

 * create ami
 * create instance from AMI
 * delete ami
 * terraform refresh should not error

If that still is the case, then we're good to go.